### DISCUSSION TITLE
IPFS resource download support

### DISCUSSION THREAD

**Discussion by stevefan1999-personal** [ppy/osu#24304] [Ideas]
This idea comes from [Bemuse](https://github.com/bemusic/bemuse), a ~IIDX~ BMS emulator that supports IPFS map import: https://bemuse.ninja/.

I found this quite interesting because this simply means we can use P2P way to distribute beatmap and other resource files like skins and I was planning to integrate this into osu so in fact, I have done some preliminary work on this, to fix the existing dotnet IPFS client and passed most of the unit tests. 

The PR could be found here: https://github.com/ipfs-shipyard/net-ipfs-core/pull/17, but unfortunately, the authors declined the PR. This makes me wonder if there is a need to create a fork of that specifically for osu.

**Comment by bdach** [comment]
I don't think we want this. Sounds like a lot of work in maintenance and upkeep which, chances are, is much more likely to be our problem in the future rather than yours down the line.

Not to mention the user consent aspect of it. Game shouldn't be passing any data along the wire without some sort of explicit or implicit consent. Which is mostly already there where online functionality is required for the game to work, but p2p is not that.

**Reply by peppy** [reply]
A more near focus would be to have this work with official servers (the goal is to stream beatmaps in file-by-file rather than require archive downloads, as the client may already have some resources), probably not using IPFS but something much simpler. Or maybe IPFS but we'll see when we cross that bridge.
